atlp-rwanda/e-commerce-furebo-32-bn
# User Logoutt-Functionality

What does this PR do?
-------------------------
This PR introduces enhancements to the logout functionality in our application. It adds robust testing for the logout endpoint to ensure its reliability and security.

Describe your changes
-------------------------
1.Added comprehensive tests for the logout functionality to cover various scenarios, including successful logout, invalid token handling, and unauthorized logout attempts.

2.Enhanced error handling in the verifyToken middleware to provide specific error messages for different JWT verification errors, such as malformed tokens.

3.Modified the Swagger documentation to include details about the logout endpoint and its usage.

How should this be manually tested?
----------------------------------------
1.Run the test suite using npm test and ensure that all logout-related tests pass without errors.

2.Manually test the logout functionality using tools like Postman or curl:
-Test logging out with a valid token.
-Test logging out without providing a token.
-Test logging out with an invalid token.

3.Verify that the Swagger documentation reflects the changes made to the logout endpoint.

What are the relevant pivotal tracker stories?
-------------------------------------------------
[Pivotal Tracker Link](https://www.pivotaltracker.com/n/projects/2703465/stories/187570730)

Screenshots (if is appropriate)
N/A

!Additional notes
-------------------
Please review the changes thoroughly, especially the additions to the logout functionality and the updated error handling in the middleware.
